shell_scripts/generate_coq_platform_packages_ci.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -e
+
+# This script generates opam/packages/*/*.dev/opam files using Coq's CI
+# project tracking info and generates $OUTPUT mentioning
+# them
+OUTPUT=${OUTPUT:=coq_platform_packages_ci.sh}
+
+TMP=`mktemp -d`
+trap "rm -rf $TMP" EXIT
+
+function log {
+  echo $1
+}
+
+function git_clone {
+  test -d $TMP/git/$1 || (log "DBG: Cloning $1 ..."; git clone $2 -b $3 --depth=1 $TMP/git/$1 >/dev/null 2>&1)
+}
+
+function test_package {
+   printf "%-52s # %s\n" "PACKAGES=\"\${PACKAGES} $1.dev\"" "$2" >> $OUTPUT
+}
+
+function skip_package {
+   printf "#%-51s # %s\n" "PACKAGES=\"\${PACKAGES} $1.dev\"" "$2" >> $OUTPUT
+}
+
+function create_opam_pinned_package {
+  local cip="$1"
+  local package="$2"
+  local REPO_VAR="${cip}_CI_GITURL"
+  local REPO=${!REPO_VAR}
+  local BRANCH_VAR="${cip}_CI_REF"
+  local BRANCH=${!BRANCH_VAR}
+  git_clone $cip ${REPO} ${BRANCH}
+  mkdir -p opam/packages/$package/$package.dev/
+  if [ -e $TMP/git/$cip/$package.opam ]; then
+    cp $TMP/git/$cip/$package.opam opam/packages/$package/$package.dev/opam
+  elif [ -e $TMP/git/$cip/*.opam ]; then
```

Comment:
Sometimes people call, in their repo, the opam file `foo.opam` instead of `coq-foo`, don't ask me why.

Yes, I thought about that, and maybe the ones in the usual repo should have precedence, and we should also look in the local repo with higher precedence... I think we should think carefully about these rule (I did not yet).